pdf/pdfcanvas.cpp

```diff
@@ -535,5 +582,123 @@ void PDFCanvas::resizeTimeout()
 void PDFCanvas::pageSizesFinished(const QList< QSizeF >& sizes)
 {
     d->pageSizes = sizes;
+    if (d->rememberWidth > 0) {
+      emit requestPageWidth(d->rememberWidth);
+    }
     layout();
+    if (d->rememberPage > 0 || d->rememberTop > 0. || d->rememberLeft > 0.) {
+      /* We wait for the layout to be done here, so pageRectangle() will
+         be available at QML level. */
+      emit requestPagePosition(d->rememberPage, d->rememberTop, d->rememberLeft);
+    }
+}
+
+/* Part related to access the local storage, as available from QML side. */
+static const QString dbConnection{"PdfSettings"};
+static QString _databasesPath()
+{
+    return QStandardPaths::writableLocation(QStandardPaths::DataLocation) +
+    QDir::separator() + QLatin1String("QML") + QDir::separator() + 
+    QLatin1String("OfflineStorage") + QDir::separator() + QLatin1String("Databases");
+}
+static void _initDatabasesPath()
+{
+    QString databasesPath = _databasesPath();
+    if (!QDir().mkpath(databasesPath))
+        qWarning() << "LocalStorage: can't create path - " << databasesPath;
+}
+
+static bool _openDatabaseSync(const QString &dbname, const QString &dbversion,
+                              const QString &dbdescription, unsigned int estimatedsize)
+{
+    _initDatabasesPath();
+
+    QCryptographicHash md5(QCryptographicHash::Md5);
+    md5.addData(dbname.toUtf8());
+    QString dbid(QLatin1String(md5.result().toHex()));
+
+    QString basename = _databasesPath() + QDir::separator() + dbid;
+
+    QSettings ini(basename + QLatin1String(".ini"), QSettings::IniFormat);
```

Comment:
Same same here, it's copied from the LocalStorage plugin of Qt. It can be removed if you think so. It's just that LocalStorage from JS always creates two files : the SQlite file and a ini file that store some meta information of the database.